### PR TITLE
Merge DotNetE2EServerApp subdirectory of azure-mobile-services-test repo into this repo

### PR DESCRIPTION
**Pull request by dhei**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@dhei__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (dihei). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: E2EServerApp/.gitignore [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+*.log
+[Bb]in
+[Dd]ebug*/
+*.lib
+*.sbr
+obj/
+[Rr]elease*/
+_ReSharper*/
+[Tt]est[Rr]esult*
+packages/
```
Comment (brettsam) [inline E2EServerApp/.gitignore]: We shouldn't need this file.

**Comment by brettsam** [comment]
We should probably follow the /E2ETest root folder pattern that the client repos are taking with this.

**Comment by dhei** [comment]
@brettsam: Agree. I will rename "/E2EServerApp" into "/e2etest".

**Pull request merged**